### ISSUE TITLE
Unknown labels in istio_requests_total metrics when the destination workload does not have sidecar

### ISSUE THREAD

**Issue by iandyh** [istio/istio#28387]
**Bug description**
```
istio_requests_total{connection_security_policy="unknown",destination_app="unknown",destination_canonical_revision="latest",destination_canonical_service="unknown",destination_service="exporter-app.caas-sentinel.svc.cluster.local",destination_service_name="exporter-app.caas-sentinel.svc.cluster.local",destination_service_namespace="unknown",destination_version="unknown",destination_workload="unknown",destination_workload_namespace="unknown",instance="100.86.8.169:15090",job="envoy-stats",namespace="istio-system",pod_name="istio-ingressgateway-bbcdd69b8-v8dd8",reporter="source",request_protocol="http",response_code="503",response_flags="UF,URX",source_app="istio-ingressgateway",source_canonical_revision="latest",source_canonical_service="istio-ingressgateway",source_version="2020-08-25-02.07.09",source_workload="istio-ingressgateway",source_workload_namespace="istio-system"}
```
Several labels are unknown. for example, destination_app, destination_service_namespace. Despite the required labels are presented.

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ X ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Metrics regarding destination should not be unknown

**Steps to reproduce the bug**
1. Make a deployment without sidecar injection
2. Send the traffic from istio-ingressgateway to the pod

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
Istio 1.7.3

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by bianpengyuan** [comment]
This has been addressed in 1.8. Although `destination_app` label is not going to be filled, since it is going to be replaced by canonical service label.

**Comment by iandyh** [comment]
@bianpengyuan Great. Thank you very much.
